File: api/domains/intelligence_hub/routes/pulse.py
"""Pulse digest API — GET /api/pulse (read-only)."""

from __future__ import annotations

from typing import Any

from fastapi import APIRouter, Query

from services.pulse_service import compute_pulse

router = APIRouter(prefix="/api", tags=["Pulse"])


@router.get("/pulse")
def get_pulse(
    window_hours: int = Query(48, ge=1, le=168),
    limit: int = Query(20, ge=1, le=100),
    domain: str | None = Query(None, description="Optional domain_key filter"),
) -> dict[str, Any]:
    payload = compute_pulse(window_hours=window_hours, limit=limit, domain_filter=domain)
    return {"success": True, "data": payload, "message": None}

File: web/src/services/api/pulse.ts
import { getApi } from './client';
import type { PulsePayload } from '@/types/pulse';

export const pulseApi = {
  async getPulse(params?: { window_hours?: number; limit?: number; domain?: string }) {
    const res = await getApi().get<{ success: boolean; data: PulsePayload; message?: string }>(
      '/api/pulse',
      { params }
    );
    return res.data;
  },
};
